test/index.test.ts

```diff
@@ -0,0 +1,110 @@
+import { expect, it, describe } from "vitest";
+import destr from "../src";
+
+describe("destr", () => {
+  it("returns the passed value if it's not a string", () => {
+    const testCases = [
+      { input: {} },
+      { input: [] },
+      { input: 123 },
+      { input: true },
+      { input: false },
+      /* eslint-disable-next-line unicorn/no-null */
+      { input: null },
+      { input: Number.POSITIVE_INFINITY },
+      { input: undefined }
+    ];
+
+    for (const testCase of testCases) {
+      expect(destr(testCase.input)).toStrictEqual(testCase.input);
+    }
+  });
+
+  it("parses string 'true' as boolean `true` case-insensitively", () => {
+    expect(destr("true")).toStrictEqual(true);
+    expect(destr("TRUE")).toStrictEqual(true);
+  });
+
+  it("parses string 'false' as boolean `false` case-insensitively", () => {
+    expect(destr("false")).toStrictEqual(false);
+    expect(destr("FALSE")).toStrictEqual(false);
+  });
+
+  it("parses string 'null' as `null`", () => {
+    /* eslint-disable unicorn/no-null */
+    expect(destr("null")).toBeNull();
+    expect(destr("NULL")).toBeNull();
+    /* eslint-enable unicorn/no-null */
+  });
+
+  it("parses string 'NaN' as `Number.NaN` case-insensitively", () => {
+    expect(destr("nan")).toBeNaN();
+    expect(destr("Nan")).toBeNaN();
+  });
+
+  it("parses string 'infinity' as `Number.POSITIVE_INFINITY` case-insensitively", () => {
+    expect(destr("infinity")).toStrictEqual(Number.POSITIVE_INFINITY);
+    expect(destr("Infinity")).toStrictEqual(Number.POSITIVE_INFINITY);
+  });
+
+  it("parses string 'undefined' as `undefined` case-insensitively", () => {
+    expect(destr("undefined")).toBeUndefined();
+    expect(destr("UNDEFINED")).toBeUndefined();
+  });
+
+  it("parses valid JSON strings", () => {
+    const testCases = [
+      { input: "{}", output: {} },
+      { input: "[]", output: [] },
+      { input: "{ \"key\": \"value\" }", output: { key: "value" } },
+      { input: "[1,2,3]", output: [1, 2, 3] }
+    ];
+
+    for (const testCase of testCases) {
+      expect(destr(testCase.input)).toStrictEqual(testCase.output);
+    }
+  });
+
+  it("prevents prototype pollution", () => {
+    const testCases = [
+      /* eslint-disable quotes */
+      { input: '{ "__proto__": {} }', output: {} },
+      { input: '{ "constructor": {} }', output: {} }
+      /* eslint-enable quotes */
+    ];
```

Comment:
I'm not sure, but I couldn't use single quotes without disabling the rule even if I used double-quotes in the string literal.

I might be missing something, or you might want to set [`avoidEscape` option](https://eslint.org/docs/latest/rules/quotes#options) to `unjs/eslint-config-unjs`.

Let me know if you want me to submit a PR to the repository.

Reply:
Can you please open an issue in eslintconfig repo to not forget?